### PR TITLE
Upgrade to latest fabric go mod packages.

### PR DESCRIPTION
**Pull request by SamYuan1990**
Nil pointer handler
Upgrade to latest fabric go mod packages.
Updated CI

### CODE REVIEW

**Comment by SamYuan1990** [comment]
@guoger , do we need to change ci to enable go mod?

```
# Go
# Build your Go project.
# Add steps that test, save build artifacts, deploy, and more:
# https://docs.microsoft.com/azure/devops/pipelines/languages/go

trigger:
- master

pool:
  vmImage: 'ubuntu-latest'

variables:
  GOBIN:  '$(GOPATH)/bin' # Go binaries path
  GOROOT: '/usr/local/go1.11' # Go installation path
  GOPATH: '$(system.defaultWorkingDirectory)/gopath' # Go workspace path
  GO111MODULE: 'on'
  modulePath: '$(GOPATH)/src/github.com/$(build.repository.name)' # Path to the module's code

steps:
- script: |
    mkdir -p '$(GOBIN)'
    mkdir -p '$(GOPATH)/pkg'
    mkdir -p '$(modulePath)'
    shopt -s extglob
    shopt -s dotglob
    mv !(gopath) '$(modulePath)'
    echo '##vso[task.prependpath]$(GOBIN)'
    echo '##vso[task.prependpath]$(GOROOT)/bin'
  displayName: 'Set up the Go workspace'

- script: |
    go version
    go mod tidy
    go test ./test -test.bench=".*"
    go build -v .
  workingDirectory: '$(modulePath)'
  displayName: 'Get dependencies, then build'
```

**Comment by guoger** [comment]
It seems that mock fabric from SDK does not actually validate anything, and by sending tx to mock peer, we are merely testing the grpc connection, which is not super helpful IMO.

could we consider #15 instead?

**Comment by SamYuan1990** [comment]
> #15

UT as unit test to keep code works. #15 looks like sample of the project. In my opinion, diff things.

**Comment by SamYuan1990** [comment]
@guoger , just rebase with latest master, pls take a review.

**Comment by SamYuan1990** [comment]
@guoger , just rebase with latest master, pls take a review.

**Comment by SamYuan1990** [comment]
currently cover function
```
proposer.CreateProposer
proposer.Start
```

**Comment by guoger** [comment]
pls split the part you bump fabric version (and switch to use new proto pkg) to a **separate PR** so we could get that merged quickly

**Comment by SamYuan1990** [comment]
> pls split the part you bump fabric version (and switch to use new proto pkg) to a **separate PR** so we could get that merged quickly

removed test file, changed pr as Upgrade to latest fabric go mod packages.

File: go.mod [hunk 1]
```diff
@@ -3,22 +3,24 @@ module github.com/guoger/stupid
...
 require (
-	github.com/Knetic/govaluate v3.0.0+incompatible // indirect
 	github.com/Shopify/sarama v1.23.1 // indirect
 	github.com/fsouza/go-dockerclient v1.4.4 // indirect
 	github.com/gogo/protobuf v1.2.1
 	github.com/golang/protobuf v1.3.2
 	github.com/hashicorp/go-version v1.2.0 // indirect
-	github.com/hyperledger/fabric v1.4.3
-	github.com/hyperledger/fabric-amcl v0.0.0-20181230093703-5ccba6eab8d6 // indirect
+	github.com/hyperledger/fabric v2.0.1+incompatible
```
Comment (guoger) [inline go.mod:11]: fabric internalized `comm` pkg in later releases. I think we could actually remove fabric from dependencies entirely (copy some code over)

**Comment by SamYuan1990** [comment]
address the comment in separate PR pls.

**Comment by guoger** [comment]
@SamYuan1990 i've enabled sign-off check in CI. Please make sure to use `git commit -s` for future contributions (and there should a blank line before sign-off line)

**Pull request merged**